File: .tmp_py4013/batch_12_dump/3332_maximum_points_tourist_can_earn.js
// LeetCode 3332 - Maximum Points Tourist Can Earn
// https://leetcode.com/problems/maximum-points-tourist-can-earn/

var maxScore = function(n, k, stayScore, travelScore) {
    let dp = new Array(n).fill(0);
    for (let day = 0; day < k; day++) {
        const ndp = new Array(n).fill(-(1 << 30));
        for (let dest = 0; dest < n; dest++) {
            let best = -(1 << 30);
            for (let src = 0; src < n; src++) {
                let val = dp[src];
                if (src === dest) val += stayScore[day][dest];
                else val += travelScore[src][dest];
                if (val > best) best = val;
            }
            ndp[dest] = best;
        }
        dp = ndp;
    }
    let ans = dp[0];
    for (let i = 1; i < n; i++) if (dp[i] > ans) ans = dp[i];
    return ans;
};

File: .tmp_rs4013/rust_check16/3332_maximum_points_tourist_can_earn.rs
struct Solution;
// LeetCode 3332 - Maximum Points Tourist Can Earn
// https://leetcode.com/problems/maximum-points-tourist-can-earn/

impl Solution {
    pub fn max_score(n: i32, k: i32, stay_score: Vec<Vec<i32>>, travel_score: Vec<Vec<i32>>) -> i32 {
        let n = n as usize;
        let k = k as usize;
        let mut dp = vec![0i32; n];
        for day in 0..k {
            let mut ndp = vec![-(1 << 30); n];
            for dest in 0..n {
                let mut best = -(1 << 30);
                for src in 0..n {
                    let mut val = dp[src];
                    if src == dest {
                        val += stay_score[day][dest];
                    } else {
                        val += travel_score[src][dest];
                    }
                    if val > best {
                        best = val;
                    }
                }
                ndp[dest] = best;
            }
            dp = ndp;
        }
        *dp.iter().max().unwrap()
    }
}

fn main() {}
